### PR TITLE
Add string rendering, remove psr/container-implementation dependency

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Review by vjik — COMMENTED** [review]
We can move code from `renderProxy` to `renderAsString` and use him instead `renderProxy`.

**Comment by devanych** [comment]
> We can move code from `renderProxy` to `renderAsString` and use him instead `renderProxy`.

Done.

**Review by vjik — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -32,7 +32,14 @@ composer require yiisoft/yii-view --prefer-dist
...
 
-The view renderer renders the view template and returns result as a `Psr\Http\Message\ResponseInterface` instance:
+There are two ways to render a view:
+
+- Return an instance of the `Yiisoft\DataResponse\DataResponse` class with deferred rendering.
+- Render immediately and return the rendered result as a string.
+
+The `Yiisoft\DataResponse\DataResponse` class is an implementation of the `Psr\Http\Message\ResponseInterface`, for
+more information about this class, see the [yiisoft/data-response](https://github.com/yiisoft/data-response) package.
+You can get an instance of a response with deferred rendering as follows:
```
Comment (samdark) [inline README.md:42]:
```suggestion
The `Yiisoft\DataResponse\DataResponse` class is an implementation of the `Psr\Http\Message\ResponseInterface`. For more information about this class, see the [yiisoft/data-response](https://github.com/yiisoft/data-response) package. You can get an instance of a response with deferred rendering as follows:
```

File: README.md [hunk 2]
```diff
@@ -55,7 +62,9 @@ $response = $viewRenderer->render('site/page', [
 ]);
 ```
 
-If a layout is set, but you need to render a view without the layout, you can use an immutable setter `withLayout()`:
+The rendering will be performed directly when calling the `getBody()` or `getData()` methods of the
+`Yiisoft\DataResponse\DataResponse` class. If a layout is set, but you need to render a view
+without the layout, you can use an immutable setter `withLayout()`:
```
Comment (samdark) [inline README.md:67]:
```suggestion
The rendering will be performed directly when calling `getBody()` or `getData()` methods of the
`Yiisoft\DataResponse\DataResponse`. If a layout is set, but you need to render a view
without the layout, you can use an immutable setter `withLayout()`:
```

File: README.md [hunk 3]
```diff
@@ -75,6 +84,21 @@ $response = $viewRenderer->renderPartial('site/page', [
 ]);
 ```
 
+To render immediately and return the rendering result as a string,
+use the `renderAsString()` and `renderPartialAsString()` methods:
```
Comment (samdark) [inline README.md:88]:
```suggestion
To render immediately and return the rendering result as a string,
use `renderAsString()` and `renderPartialAsString()` methods:
```

File: README.md [hunk 4]
```diff
@@ -88,6 +112,14 @@ see description of the [yiisoft/aliases](https://github.com/yiisoft/aliases) pac
 
 If the view renderer is used in a controller, you can either specify controller name explicitly using
 `withControllerName()` or determine name automatically by passing a controller instance to `withController()`.
+Automatically, the name of the controller is determined as follows:
```
Comment (samdark) [inline README.md:115]:
```suggestion
In this case the name is determined as follows:
```

File: src/ViewRenderer.php [hunk 5]
```diff
@@ -32,10 +34,14 @@
...
+ * ViewRenderer renders the view.
  *
- * @psalm-import-type MetaTagsConfig from MetaTagsInjectionInterface
- * @psalm-import-type LinkTagsConfig from LinkTagsInjectionInterface
+ * If the {@see ViewRenderer::render()} or {@see ViewRenderer::renderPartial()} methods are called,
+ * an instance of the {@see DataResponse} is returned, with deferred rendering. Rendering will
+ * occur when calling {@see DataResponse::getBody()} or {@see DataResponse::getData()}.
+ *
+ * If the {@see ViewRenderer::renderAsString()} or {@see ViewRenderer::renderPartialAsString()} methods are called,
+ * the rendering will occur immediately and the string result of the rendering will be returned.
```
Comment (samdark) [inline src/ViewRenderer.php:44]:
```suggestion
 * If {@see ViewRenderer::render()} or {@see ViewRenderer::renderPartial()} methods are called,
 * an instance of {@see DataResponse} is returned. It supports deferred rendering that
 * occurs when calling {@see DataResponse::getBody()} or {@see DataResponse::getData()}.
 *
 * If the {@see ViewRenderer::renderAsString()} or {@see ViewRenderer::renderPartialAsString()} methods are called,
 * the rendering will occur immediately and the string result of the rendering will be returned.
```

File: src/ViewRenderer.php [hunk 6]
```diff
@@ -90,42 +96,34 @@ public function getViewPath(): string
     }
 
     /**
-     * Renders a view returns a response instance.
+     * Returns a response instance {@see DataResponse} with deferred rendering {@see renderAsString()}.
```
Comment (samdark) [inline src/ViewRenderer.php:99]:
```suggestion
     * Returns a response instance {@see DataResponse} that supports deferred rendering {@see renderAsString()}.
```

File: tests/ViewRendererTest.php [hunk 7]
```diff
@@ -65,6 +69,17 @@ public function testRenderWithFullPathLayout(): void
         $this->assertSame('<html><body><b>donatello</b></body></html>', (string) $response->getBody());
     }
 
+    public function testRenderAsStringWithFullPathLayout(): void
```
Comment (samdark) [inline tests/ViewRendererTest.php:72]:
```suggestion
    public function testRenderAsStringWithAbsoluteLayoutPath(): void
```

**Comment by samdark** [comment]
👍

**Pull request merged**